src/package.rs

```diff
@@ -429,73 +436,139 @@ impl FromStr for PackageType {
 
 impl fmt::Display for PackageType {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", serde_typename::to_str(self).unwrap_or_default())
+        match serde_typename::to_str(self) {
+            Ok(value) => f.write_str(value),
+            Err(_error) => unreachable!(),
+        }
     }
 }
 
-impl Default for PackageType {
-    fn default() -> Self {
-        Self::Impl
+#[test]
+fn can_check_publishable() {
+    assert!(PackageType::Lib.is_publishable());
+    assert!(PackageType::Api.is_publishable());
+    assert!(!PackageType::Impl.is_publishable());
+}
+
+#[test]
+fn can_check_compilable() {
+    assert!(PackageType::Lib.is_compilable());
+    assert!(PackageType::Api.is_compilable());
+    assert!(!PackageType::Impl.is_compilable());
+}
+
+#[test]
+fn can_default_package_type() {
+    assert_eq!(PackageType::default(), PackageType::Impl);
+}
+
+#[test]
+fn can_parse_package_type() {
+    let types = [PackageType::Lib, PackageType::Api, PackageType::Impl];
+    for typ in &types {
+        let string = typ.to_string();
+        let parsed: PackageType = string.parse().unwrap();
+        assert_eq!(parsed, *typ);
     }
 }
 
 /// A `buffrs` package name for parsing and type safety
-#[derive(Clone, Hash, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Clone, Hash, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord, Debug)]
 #[serde(try_from = "String", into = "String")]
 pub struct PackageName(String);
 
-impl TryFrom<String> for PackageName {
-    type Error = miette::Report;
+/// Errors that can be generated parsing [`PackageName`][], see [`PackageName::new()`][].
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum PackageNameError {
```

Comment:
```suggestion
enum PackageNameError {
```

Reply:
And please, if possible, derive miette diagnostics

Reply:
I strongly disagree with making this private.